### PR TITLE
Patch repo name

### PR DESCRIPTION
**Pull request by SeanTConrad**
## Why are these changes needed?

Getting the repo name and URL fails for some use cases. Abstracted out to a function with a unit test. WIP. The unit test needs all the use cases we know about and want to support. The new function is now used to get repo name and branch

## Checks

- [ ] I've added tests (if relevant) corresponding to the changes introduced in this PR.
- [ ] I've made sure all auto checks have passed.

### CODE REVIEW

File: src/verinfast/utils/utils.py [hunk 1]
```diff
@@ -147,3 +147,31 @@ def log(self, msg, tag=None, display=False, timestamp=True):
             f.write(output+"\n")
         if display or self.debug:
             print(output)
+
+
+# Returns a tuple of the repo name and the repo url from the original url
+def get_repo_name_url_and_branch(repo_url: str):
+    # match = re.search(r"([^/]*\.git.*)", repo_url) ^.*?/(.*)
```
Comment (aylusltd) [inline src/verinfast/utils/utils.py:154]: Can we remove this? it's really confusing to have two commented out lines that are basically the same thing.

Comment (SeanTConrad) [reply src/verinfast/utils/utils.py]: Fixed. I also set the agent to use the new function

File: src/verinfast/agent.py [hunk 2]
```diff
@@ -583,16 +580,10 @@ def scanRepos(self):
...
-                    match = re.search(r"([^/]*\.git.*)", repo_url)
-                    if match:
-                        repo_name = match.group(1)
-                    else:
-                        repo_name = repo_url.rsplit('/', 1)[-1]
-                    if "@" in repo_name and re.search(r"^.*@.*\..*:", repo_url):
-                        repo_url = "@".join(repo_url.split("@")[0:2])
-                    elif "@" in repo_name:
-                        repo_url = repo_url.split("@")[0]
+                    repo_name, repo_url, branch = get_repo_name_url_and_branch(repo_url)
```
Comment (aylusltd) [inline src/verinfast/agent.py:583]: it's really bad form to return tuples. Technically allowed, but better to return dicts. that way order doesn't matter and each thing is named.

File: tests/test_repo_name.py [hunk 3]
```diff
@@ -0,0 +1,117 @@
+from verinfast.utils.utils import get_repo_name_url_and_branch
+
+url_list = [
```
Comment (aylusltd) [inline tests/test_repo_name.py:3]: prefer to move this to a separate file in fixtures.

File: src/verinfast/utils/utils.py [hunk 4]
```diff
@@ -147,3 +147,36 @@ def log(self, msg, tag=None, display=False, timestamp=True):
...
+    # The regex pattern captures the entire URL
+    # and the repository name after the last '/'
+
+    repo_url = repo_url.strip()
+    branch = None
+
+    split_url = repo_url.split("@")
+
+    # HTTPS URLs
+    if repo_url.startswith("https://") or repo_url.startswith("http://"):
```
Comment (aylusltd) [inline src/verinfast/utils/utils.py:163]: basic auth can be used with html, e.g.

https://user:password@my.githost.com/repo@branch

File: src/verinfast/agent.py (hunk 2 above)
Comment (SeanTConrad) [reply src/verinfast/agent.py]: Fixed

File: tests/test_repo_name.py (hunk 3 above)
Comment (SeanTConrad) [reply tests/test_repo_name.py]: fixed

File: src/verinfast/utils/utils.py (hunk 4 above)
Comment (SeanTConrad) [reply src/verinfast/utils/utils.py:163]: added to unit test and then added code to handle.

**Review by aylusltd — APPROVED** [review]

**Pull request merged**